homeassistant/components/airly/__init__.py

```diff
@@ -98,20 +134,16 @@ def __init__(self, session, api_key, latitude, longitude):
             standards = measurements.current["standards"]
 
             if index["description"] == NO_AIRLY_SENSORS:
-                _LOGGER.error("Can't retrieve data: no Airly sensors in this area")
-                return
+                raise UpdateFailed("Can't retrieve data: no Airly sensors in this area")
             for value in values:
-                self.data[value["name"]] = value["value"]
+                data[value["name"]] = value["value"]
             for standard in standards:
-                self.data[f"{standard['pollutant']}_LIMIT"] = standard["limit"]
-                self.data[f"{standard['pollutant']}_PERCENT"] = standard["percent"]
-            self.data[ATTR_API_CAQI] = index["value"]
-            self.data[ATTR_API_CAQI_LEVEL] = index["level"].lower().replace("_", " ")
-            self.data[ATTR_API_CAQI_DESCRIPTION] = index["description"]
-            self.data[ATTR_API_ADVICE] = index["advice"]
-            _LOGGER.debug("Data retrieved from Airly")
-        except asyncio.TimeoutError:
-            _LOGGER.error("Asyncio Timeout Error")
+                data[f"{standard['pollutant']}_LIMIT"] = standard["limit"]
+                data[f"{standard['pollutant']}_PERCENT"] = standard["percent"]
+            data[ATTR_API_CAQI] = index["value"]
+            data[ATTR_API_CAQI_LEVEL] = index["level"].lower().replace("_", " ")
+            data[ATTR_API_CAQI_DESCRIPTION] = index["description"]
+            data[ATTR_API_ADVICE] = index["advice"]
         except (ValueError, AirlyError, ClientConnectorError) as error:
```

Comment:
We should only wrap the code in `try…except` that can raise. So in this case just the `await measurements.update()` part.

Reply:
Done.